arm64/amd64 inconsistency in Stat_t.Dev type

For some reason, the `Stat_t.Dev` type differs between amd64 and arm64:
- https://github.com/golang/go/blob/83c4e533bcf71d86437a5aa9ffc9b5373208628c/src/syscall/ztypes_linux_amd64.go#L102
- https://github.com/golang/go/blob/83c4e533bcf71d86437a5aa9ffc9b5373208628c/src/syscall/ztypes_darwin_arm64.go#L66

The consequence is that opni won't compile in ARM architectures:
```
pkg/update/patch/client/client.go:116:22: cannot use stat.Dev (variable of type int32) as uint64 value in assignment
pkg/update/patch/client/client.go:127:22: invalid operation: stat.Dev == pluginDirDevice (mismatched types int32 and uint64)
...
Error: running "go build -v=true -buildmode=archive -trimpath -tags=nomsgpack ./..." failed with exit code 1
running "go build -v=true -buildmode=archive -trimpath -tags=nomsgpack ./..." failed with exit code 1

```
 I know that we don't plan on supporting anything other than Linux, but I wonder if it makes sense to add a semi-redundant cast to these two places:
- https://github.com/rancher/opni/blob/6747c16360626f64ac7ce32d349285dddcf3c7fb/pkg/update/patch/client/client.go#L116
- https://github.com/rancher/opni/blob/6747c16360626f64ac7ce32d349285dddcf3c7fb/pkg/update/patch/client/client.go#L127

### Machine info:
```bash
$ uname -a
Darwin suse.local 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:53 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6020 arm64

$ go version
go version go1.20.6 darwin/arm64

```

--- a/pkg/update/patch/client/client.go
+++ b/pkg/update/patch/client/client.go
@@ -113,7 +113,9 @@ func findTempDirBase(baseFs afero.Fs, pluginDir string) (string, error) {
 	if sys := pluginDirInfo.Sys(); sys != nil {
 		if stat, ok := sys.(*syscall.Stat_t); ok {
 			deviceInfoAvailable = true
-			pluginDirDevice = stat.Dev
+			// This cast is necessary for arm64 compilation.
+			// See https://github.com/rancher/opni/issues/1614
+			pluginDirDevice = uint64(stat.Dev)
 		}
 	}
 
@@ -124,7 +126,9 @@ func findTempDirBase(baseFs afero.Fs, pluginDir string) (string, error) {
 			if info, err := af.Stat(candidate); err == nil {
 				if sys := info.Sys(); sys != nil {
 					if stat, ok := sys.(*syscall.Stat_t); ok {
-						if stat.Dev == pluginDirDevice {
+						// This cast is necessary for arm64 compilation.
+						// See https://github.com/rancher/opni/issues/1614
+						if uint64(stat.Dev) == pluginDirDevice {
 							return candidate, nil
 						}
 					}

